src/System.Net.HttpListener/tests/HttpListenerAuthenticationTests.cs

```diff
@@ -430,6 +450,24 @@ public void Realm_SetDisposed_ThrowsObjectDisposedException()
             return clientTask.Result;
         }
 
+        public async Task<HttpResponseMessage> AuthenticationFailureAsyncContext(HttpClient client, HttpStatusCode errorCode)
+        {
+            Task<HttpResponseMessage> clientTask = client.GetAsync(_factory.ListeningUrl);
+            Task<HttpListenerContext> serverTask = _listener.GetContextAsync();
+
+            // The client task should complete first - the server should send a 401 response.
```

Comment:
I think you're right: It shouldn't within the context of this class (while `_listener` is not disposed). 

I'm not very familiar with HttpListener but from the code investigation, once the bound handle is disposed all corresponding IOCP should also be aborted which means the kernel operation will eventually complete. (If proper in-process cleanup is a concern: I'd expect all TPL objects on the managed side will also get eventually GCed as they're not rooted anymore.)

Not my priority but it would be simpler to test if the async operations are also canceled at that point. If not, it would be nice to have as it would be a GetContextAsync with a cancellation token.